src/posts/flox/index.md

```diff
@@ -0,0 +1,181 @@
+
+GroupBy is a term used for a very common analysis pattern commonly called "split-apply-combine" ([Wickham, 2011](https://www.jstatsoft.org/article/view/v040i01)) wherein an analyst
+
+- _Splits_ a dataset into groups (e.g. counties),
+- _Applies_ a transformation to each group of data (here a reduction like `.mean`)
+- _Combines_ the results of `apply` to form a new dataset
+
+For this problem we will split the dataset into counties, apply the `mean`, and then combine the results back.
+
+With [Xarray](https://docs.xarray.dev/en/stable/user-guide/groupby.html), this would look like
+
+```python
+dataset.groupby(counties).mean()
+```
+
+However Xarray's default algorithm is to split the dataset in to groups by indexing, and then applying the reduction as a simple for-loop over groups. This approach doesn't work very well for large distributed problems.
+
+![Xarray default groupby](https://flox.readthedocs.io/en/latest/_images/new-split-apply-combine-annotated.svg)
+
+## Enter flox.
+
+`flox` solves a long-standing problem in the Pangeo array computing ecosytem of computing GroupBy reductions. It implements a parallel groupby algorithm (using a tree reduction) to substantially improve performance of groupby reductions with dask.
+
+![flox default map-reduce groupby](https://flox.readthedocs.io/en/latest/_images/new-map-reduce-reindex-True-annotated.svg)
+
+- Specifically, `flox` speeds up [reduction methods](https://flox.readthedocs.io/en/latest/aggregations.html) like `groupby(...).mean()`, `groupby(...).max()`, etc, but not `groupby.map`.
+- `flox` also significantly speeds up groupby reductions with pure numpy arrays using optimized implementations in the [`numpy-groupies` package](https://github.com/ml31415/numpy-groupies).
+- `flox` allows more complicated groupby operations such as lazy grouping by a dask array, and grouping by multiple variables. Use `flox.xarray.xarray_reduce` for [these operations](https://flox.readthedocs.io/en/latest/xarray.html). Xarray currently only supports grouping by a single numpy variable.
+
+See [here](https://flox.readthedocs.io/en/latest/intro.html) for short examples.
+
+### How do I use it?
+
+Run `mamba install flox` and `xarray>=2022.06.0` will use it by default for `.groupby`, `.groupby_bins`, and `.resample`!
+
+A lot of effort was spent in ensuring backwards compatibility, so your workloads should only work better. Let us know if it [does not](https://github.com/pydata/xarray/issues)
+
+## Demo
+
+### Load NWM data
+
+```python
+import flox  # make sure its available
+import fsspec
+import numpy as np
+import rioxarray
+import xarray as xr
+
+ds = xr.open_zarr(
+    fsspec.get_mapper("s3://noaa-nwm-retrospective-2-1-zarr-pds/rtout.zarr", anon=True),
+    consolidated=True,
+)
+```
+
+Each field in this dataset is big!
+
+```{python}
+ds.zwattablrt
+```
+
+<RawHTML filePath='/posts/flox/zwattablrt-repr.html' />
```

Comment:
for the HTML repr to work, the HTML file needs to reside under `/public` directory. Let's move the actual HTML files into the `/public/posts/flox` directory instead of `/src/posts/flox` directory